shelley/chain-and-ledger/formal-spec/update.tex

```diff
@@ -0,0 +1,396 @@
+\section{Updates}
+\label{sec:update}
+
+
+The $\mathsf{UPDATE}$ transition is responsible for federated governance model in Shelly.
+This includes the ability to change the protocol parameters and also handles software
+updates.
+
+The application versions behave much like a protocol parameter, but because adoption
+for them must occur as soon as consensus is met, they are handled separately.
+Therefore the $\mathsf{UPDATE}$ is the combination of the
+Therefore the $\mathsf{PPUP}$ transition for the protocol parameters and the
+the $\mathsf{AVUP}$ transition for the application versions.
+
+Note that the signature for the keys in the proposal will be checked in the
+$\mathsf{UTXOW}$ transition.
+
+
+\begin{figure}[htb]
+  \emph{Protocol Parameter Update environment}
+  \begin{equation*}
+    \PPUpdateEnv =
+    \left(
+      \begin{array}{r@{~\in~}lr}
+        \var{slot} & \Slot & \text{current slot}\\
+        \var{dms} & \VKeyGen\mapsto\VKey & \text{genesis key delegations} \\
+      \end{array}
+    \right)
+  \end{equation*}
+  %
+  \emph{Protocol Parameter Update transitions}
+  \begin{equation*}
+    \_ \vdash
+    \var{\_} \trans{ppup}{\_} \var{\_}
+    \subseteq \powerset (\PPUpdateEnv \times \PPUpdate \times \PPUpdate \times \PPUpdate)
+  \end{equation*}
+  %
+  \caption{Protocol Parameter Update transition-system types}
+  \label{fig:ts-types:pp-update}
+\end{figure}
+
+\begin{figure}[htb]
+  \begin{equation}\label{eq:pp-update-Empty}
+    \inference[PP-Update-Empty]
+    {
+      \var{pup} = \emptyset
+    }
+    {
+      \begin{array}{l}
+        \var{slot}\\
+        \var{dms}\\
+      \end{array}
+      \vdash \var{pup_s}\trans{ppup}{pup}\var{pup_s}
+    }
+  \end{equation}
+
+  \nextdef
+
+  \begin{equation}\label{eq:update-nonempty}
+    \inference[PP-Update-Nonempty]
+    {
+      \var{pup}\neq\emptyset
+      &
+      \dom{pup}\subseteq\dom{dms}
```

Comment:
It seems that we use `dms` here only for its domain. Would it make sense to pass in the environment only what's needed, i.e. the allowed proposers (?).

Reply:
I'm going to pass on this for now, if that's okay. once all the dust settles, we can think about if this optimization is worth it.

Reply:
Sure.